File: leetcode-no6.py
class Solution2(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        node = 2*numRows - 2
        if node == 0:
            node = 1
        add = node - len(s)%node
        s += " "*add
        slide = []
        for i in range(0,len(s),node):
            slide.append(s[i:i+node])
        result = ""
        for i in range(numRows):
            for h in slide:
                result += h[i]
                if i > 0 and i < numRows-1:
                    result += h[node - i]
        return result.replace(" ",'')

class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        result = ['']*numRows
        node = numRows - 1
        if node == 0:
            return s
        length = len(s)
        change = 0
        for i in range(0,length):
            if change == 0:
                result[i%node] += s[i]
            else:
                result[node-i%node] += s[i]
            if i%node == node -1 and change == 0:
                change = 1
            elif i%node == node - 1 and change == 1:
                change = 0
        output = ''
        for i in result:
            output += i
        return output
t = Solution()
print(t.convert("PAYPALISHIRING",3))

File: 0000_0099/leetcode-no6.cpp
#include <iostream>
#include <String>
#include <cstdio>
#include <vector>
using namespace std;
class Solution {
public:
    string convert(string s, int numRows) {
        vector<string> result;
        result.resize(1000,"");
        int node = numRows - 1;
        if (node == 0)
            return s;
        int length = s.length();
        int change = 0;
        for (int i = 0;i<length;i++){
            if (change == 0){
                result[i%node] += s[i];
            } else {
                result[node-i%node] += s[i];
            }
            if ((i % node == node - 1) && change == 0){
                change = 1;
            } else if ((i%node == node - 1) && change == 1) {
                change = 0;
            }
        }
        string output = "";
        for (auto j : result){
            output += j;
        }
        return output;
    }
};

int main(){
    Solution t;
    string ans = t.convert("PAYPALISHIRING",3);
    cout << ans << '\n';
    return 0;
}



